devito/finite_differences/coefficients.py

```diff
@@ -224,6 +224,33 @@ def generate_subs(i):
         rules = {}
         for i in self.coefficients:
             rules.update(generate_subs(i))
+        return rules
+
+    def update_rules(self, obj):
+        """Update the specified rules to reflect staggering in an equation"""
+        # Determine which 'rules' are expected
+        sym = get_sym(self._function_list)
+        terms = obj.find(sym)
+        args_expected = filter_ordered(term.args[1:] for term in terms)
+        args_expected_dim = [(arg[0], arg[1], retrieve_dimensions(arg[2])[0])
```

Comment:
you're often using `arg[0], arg[1], ...` but what are these objects? I almost never use explicit indexing into tuples/lists throughout devito. I think you're after a `namedtuple` here: https://docs.python.org/3/library/collections.html#collections.namedtuple (if yes, `git grep namedtuple` for examples oh how it's used in devito)

Reply:
What are `arg[0]` etc. exactly? Are they not already available from some object we have access to?